Source/WebCore/rendering/RenderTheme.cpp

```diff
@@ -1321,83 +1321,246 @@ auto RenderTheme::colorCache(OptionSet<StyleColorOptions> options) const -> Colo
+    // Text in visited links. For light backgrounds, traditionally purple.
     case CSSValueVisitedtext:
         return SRGBA<uint8_t> { 85, 26, 139 };
-    case CSSValueActiveborder:
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-activetext
+    // Text in active links. For light backgrounds, traditionally red.
+    case CSSValueActivetext:
+    case CSSValueWebkitActivelink: // Non-standard addition.
+        return Color::red;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonface
+    // The face background color for push buttons.
+    case CSSValueButtonface:
+        return Color::lightGray;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttontext
+    // Text on push buttons.
+    case CSSValueButtontext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-buttonborder
+    // The base border color for push buttons.
+    case CSSValueButtonborder:
+        return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-field
+    // Background of input fields.
+    case CSSValueField:
         return Color::white;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-fieldtext
+    // Text in input fields.
+    case CSSValueFieldtext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-highlight
+    // Background of selected text, for example from ::selection.
+    case CSSValueHighlight:
+        return SRGBA<uint8_t> { 181, 213, 255 };
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-highlighttext
+    // Text of selected text.
+    case CSSValueHighlighttext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-selecteditem
+    // Background of selected items, for example a selected checkbox.
+    case CSSValueSelecteditem:
+        return Color::lightGray;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-selecteditemtext
+    // Text of selected items.
+    case CSSValueSelecteditemtext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-mark
+    // Background of text that has been specially marked (such as by the HTML mark element).
+    case CSSValueMark:
+        return Color::yellow;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-marktext
+    // Text that has been specially marked (such as by the HTML mark element).
+    case CSSValueMarktext:
+        return Color::black;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-graytext
+    // Disabled text. (Often, but not necessarily, gray.)
+    case CSSValueGraytext:
+        return Color::darkGray;
+
+    // https://drafts.csswg.org/css-color-4/#valdef-system-color-accentcolor
+    // Background of accented user interface controls.
+    case CSSValueAccentcolor:
+        return SRGBA<uint8_t> { 0, 122, 255 };
```

Comment:
No idea what an "accented" user interface control is. Perhaps a concept that does not exist on Apple platforms?

Reply:
This would be the system accent color on Apple platforms, which is configurable on macOS using `Appearance > Accent Color` in System Settings.

Something we would want to force to a single value even when implementing the platform-specific version, to avoid fingerprinting risk.

Reply:
Yeah, we have an existing webkit specific named color, CSSValueAppleSystemControlAccent, which uses this same hard coded value when not in the special "allow system UI" mode Tim mentioned above.

In a follow up, we should consider making CSSValueAccentcolor have the same behavior as CSSValueAppleSystemControlAccent.